crates/ruff_python_ast/src/visitor/transformer.rs

```diff
@@ -0,0 +1,732 @@
+use crate::{
+    self as ast, Alias, Arguments, BoolOp, CmpOp, Comprehension, Decorator, ElifElseClause,
+    ExceptHandler, Expr, ExprContext, Keyword, MatchCase, Operator, Parameter, Parameters, Pattern,
+    PatternArguments, PatternKeyword, Stmt, TypeParam, TypeParamTypeVar, TypeParams, UnaryOp,
+    WithItem,
+};
+
+/// A trait for transforming ASTs. Visits all nodes in the AST recursively in evaluation-order.
```

Comment:
Is it important that this modifies the tree in evaluation order or could we use pre-order instead (I was very surprised when I learned that our default visitor visits in evaluation and not pre-order)